src/DI/ContainerBuilder.php

```diff
@@ -411,6 +417,17 @@ private function resolveImplement(ServiceDefinition $def, $name)
 	}
 
 
+	private function getTypeHintForParameter(\ReflectionParameter $param)
+	{
+		if (PHP_VERSION_ID >= 70000 && $param->hasType()) {
+			$type = $param->getType();
+			return $type->isBuiltin() ? (string) $type : '\\' . $type;
+		} else {
+			return $param->isArray() ? 'array' : PhpReflection::getPropertyType($param);
```

Comment:
Functionality for 5.x stays the same as it was before; the else branch is [exactly the same as it used to be](https://github.com/nette/di/blob/master/src/DI/ContainerBuilder.php#L395). I'm honestly not sure if PR for PHP7 features should change the functionality on PHP 5.x.

What do you think?